core/codenvy-hosted-machine-token-service/src/main/java/com/codenvy/service/machine/token/MachineTokenRegistry.java

```diff
@@ -0,0 +1,78 @@
+/*
+ *  [2012] - [2016] Codenvy, S.A.
+ *  All Rights Reserved.
+ *
+ * NOTICE:  All information contained herein is, and remains
+ * the property of Codenvy S.A. and its suppliers,
+ * if any.  The intellectual and technical concepts contained
+ * herein are proprietary to Codenvy S.A.
+ * and its suppliers and may be covered by U.S. and Foreign Patents,
+ * patents in process, and are protected by trade secret or copyright law.
+ * Dissemination of this information or reproduction of this material
+ * is strictly forbidden unless prior written permission is obtained
+ * from Codenvy S.A..
+ */
+package com.codenvy.service.machine.token;
+
+import com.google.common.collect.HashBasedTable;
+import com.google.common.collect.Table;
+
+import org.eclipse.che.api.core.NotFoundException;
+
+import javax.inject.Singleton;
+
+import static java.lang.String.format;
+import static org.eclipse.che.commons.lang.NameGenerator.generate;
+
+/**
+ *
+ * Table-based storage of machine security tokens.
+ * Table rows is workspace Id's, columns - user Id's.
+ *
+ * @author Max Shaposhnik (mshaposhnik@codenvy.com)
+ */
+@Singleton
+public class MachineTokenRegistry {
```

Comment:
singleton component shared through the web application, what are you talking about?